pkg/cloud/scope/cluster.go

```diff
@@ -395,3 +396,8 @@ func (s *ClusterScope) Partition() string {
 func (s *ClusterScope) AdditionalControlPlaneIngressRules() []infrav1.IngressRule {
 	return s.AWSCluster.Spec.NetworkSpec.DeepCopy().AdditionalControlPlaneIngressRules
 }
+
+// UnstructuredControlPlane returns the unstructured object for the control plane, if any.
```

Comment:
This should state the expectation if none is set yet (see below comment where you return `{}` instead of `nil`).

Reply:
fixed the comment, thanks